app/controllers.js

```diff
@@ -56,18 +56,20 @@ Controller.prototype.handleApiAddProject = function(req, res) {
 
 // JSON data of a project
 Controller.prototype.handleApiEditProject = function(req, res) {
-    let updatedProject = new Project(req.body.name, req.body.description);
-    updatedProject.setLocation(req.body.location);
-    updatedProject.setPhase(req.body.phase);
-    updatedProject.setId(req.params.projectId);
+    var project = projectCache.getById(req.params.projectId);
 
-    if(!updatedProject.id) {
-        log.debug('Error updating project. Project ID is missing');
+    if(!project.id) {
```

Comment:
if `req.params.projectId` is unknown, the `project` will be undefined. Trying to access `project.id` might throw an error. Might be worth checking `req.params.projectId` here instead